Add namespace functionality to mapping example

brianabouchard opened the pull request:
These modifications add the ability to use a namespace with the mapping example and update the documentation with instructions on how to use this new functionality. 

Additionally, added a clarification to the build instructions to ensure users are building the humble branch when appropriate.

brianabouchard committed:
Revisions to lidar example to allow for use of namespaces

- remapped topics to switch to relative from absolute in all launch files and config files
- added namespace argument in all launch files
- created new launch file for rviz that directly pulls in rviz config
- changed rviz frame rate config to 5 from 30 based on issues in testing

brianabouchard committed:
Replaces node name with wildcard syntax in yaml

Replacement needed to enable function with namespace

brianabouchard committed:
Add detailed comments to launch files for clarity for new users

brianabouchard committed:
Revised lidar readme to include instructions for namespaces

- Also clarified additional instructions including rviz install instructions.

brianabouchard committed:
Add humble branch parameter to build instructions

alsora commented on create3_lidar/README.md:
````diff
@@ -42,18 +42,25 @@ Note that all STLs of brackets referenced in this example are found in our [crea
 
 ```bash
-ros2 launch create3_lidar slam_toolbox_launch.py
+ros2 launch create3_lidar sensors_launch.py namespace:='ROBOTNAMESPACE'
````
minor: rather than showing the example with a namespace and having the description above saying to replace/remove it, I would rather do something like this

```
Run the sensors launch script, which includes the LIDAR driver and transform from the laser scan to the robot.

ros2 launch create3_lidar slam_toolbox_launch.py


If your robot is using a namespace, you should add namespace:='ROBOTNAMESPACE' to the previous command, where ROBOTNAMESPACE is the namespace of your robot
```

I think that it's easier to read / understand.
The same comment applies to other parts of the PR.

alsora commented on create3_lidar/README.md:
````diff
@@ -62,36 +69,49 @@ There may be some warnings and errors on startup, but the following message will
 
-On your computer, clone and build this, and source the setup shell scripts.
+```
+sudo apt install ros-humble-rviz2
````
this command installs everything ROS-related and requires that you already setup ROS 2 debian sources.
Should we point this out?

shamlian replied:
At this point, the user should already have ROS 2 installed, right? Doesn't ros-humble-desktop include rviz2? Maybe this should be under an "if necessary."

alsora commented on create3_lidar/README.md:
```diff
@@ -62,36 +69,49 @@ There may be some warnings and errors on startup, but the following message will
 
-On your computer, start rviz2.
+
+Run the sensors launch script, which includes the LIDAR driver and transform from the laser scan to the robot. Replace `ROBOTNAMESPACE` with the namespace of your robot. If you are not using a namespace, you may delete `namespace:='ROBOTNAMESPACE'` from the command:
```
why do we need to run this on the laptop?

alsora commented on create3_lidar/README.md:
````diff
@@ -62,36 +69,49 @@ There may be some warnings and errors on startup, but the following message will
-rviz2
+ros2 launch create3_lidar rviz_launch.py namespace:='ROBOTNAMESPACE'
 ```
+The rviz2 configuration this command imports will configure rviz2 to subscribe to the laser, the occupancy map, and display the `base_footprint` tf frame that the laser is building off of from the map frame.
````
We should just run rviz specifying the configuration, rather than launching the whole lidar TF thing.

alsora commented on create3_lidar/config/mapper_params_online_async.yaml:
```diff
@@ -1,4 +1,5 @@
-slam_toolbox:
+# slam_toolbox
+/**:
```
keep the name of the node, otherwise there may be conflicts between parameters.
see here as an example https://github.com/iRobotEducation/create3_sim/blob/main/irobot_create_common/irobot_create_common_bringup/config/hazard_vector_params.yaml#L2-L3

alsora commented:
I think that we should rename `create3_lidar` into `create3_lidar_slam`

brianabouchard committed:
Revised namespace use instructions and correction to purpose of rviz launch file.

brianabouchard committed:
Include node name in yaml after wildcard

brianabouchard committed:
Changed package name to create3_lidar_slam

shamlian commented on create3_lidar_slam/README.md:
```diff
@@ -42,19 +42,27 @@ Note that all STLs of brackets referenced in this example are found in our [crea
+On the SBC, clone and build the [create3_examples repository](https://github.com/iRobotEducation/create3_examples). 
 
-Start the sensors launch script, which includes the LIDAR driver and transform from the laser scan to the robot:
+Source the setup shell scripts in **EVERY** terminal you open:
```
til there is a difference between "sourcing" and "executing"

shamlian reviewed (commented):
Looks great, thank you for this contribution!

brianabouchard committed:
Remove rviz2 install instructions

- confirmed rviz2 installs when ros-humble-desktop is installed and create docs site provides desktop installation command for humble.
